### PR TITLE
Strip type arguments when checking types

### LINKED ISSUE #6133: Invalid type abbreviation delcaration allowed
```fsharp
type Oops = Span<int> * Span<int>

let f (x: Oops) = ()

let failsAsIntended (x: Span<int> * Span<int>) = ()
```

`failsAsIntended` will fail to compile, since you can't tuple a Span. But `f` and the `Oops` declaration do compile.

### PR DESCRIPTION
**Pull request by cartermp**
Fixes #6133 , #6104 

- [ ] Add tests

### CODE REVIEW

File: src/fsharp/PostInferenceChecks.fs [hunk 1]
```diff
@@ -324,7 +324,7 @@ let rec CheckTypeDeep (cenv: cenv) ((visitTy,visitTyconRefOpt,visitAppTyOpt,visi
             | _ -> ())
     | _ -> ()
     
-    let ty = stripTyparEqns ty 
+    let ty = stripTyEqns g ty |> stripTyparEqns
```
Comment (TIHan) [inline src/fsharp/PostInferenceChecks.fs]:
```suggestion
    let ty = 
        if isTyparTy g ty then
            stripTyparEqns ty
        else
            stripTyEqns g ty
```
The error in CI is caused by this. Perhaps combining both stripTyEqns and stripTyparEqns may not be wise to do.

**Comment by cartermp** [comment]
Weird that I could build a compiler and install it into VS (and have it work) yet CI would fail like this

**Comment by cartermp** [comment]
Whelp, I'm lost. This fsharpqa test- FamANDAssembly_NoIVT - is unpassable, even though it's unrelated to my change.

**Comment by cartermp** [comment]
#7953 is better

**Pull request closed without merging**